File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 6 - Polynomial Regression/Regression_Template/regression_template.py
# Regression Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values # upperbound not included. We want to keep in matrix form not a vector for indepednent variable??
y = dataset.iloc[:, 2].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0) 


# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting Polynomial Regression to the dataset
# Regressor would be the nonlinear regressor object. 

# Predict new result with Polynomial Regression
y2_pred = regressor.predict(6.5)


# Visualize the Polynomial Regression Results
plt.scatter(X,y, color = 'red')
plt.plot(X_grid, regressor.predict(X), color = 'blue')
plt.title('Truth or Bluff (Regression Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Visualize the Polynomial Regression Results for higher resolution
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid),1)
plt.scatter(X,y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Regression Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 6 - Polynomial Regression/Regression_Template/regression_template.R
# Polynomial Regression

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
'''library(caTools)
set.seed(123)
split = sample.split(dataset$DependentVariable, SplitRatio = 1)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)'''


# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)


# Fitting to Polynomial
#Create Regression model

# predit Polynomial Regression
y_pred_poly = predict(regressor, data.frame(Level = 6.5))

# Plot Polynomial
library(ggplot2)
ggplot() + 
  geom_point(aes(x = dataset$Level, y=dataset$Salary), colour = 'red') + 
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)), colour = 'blue') +
  ggtitle('Truth or Bluff (Regression model)') +
  xlab('Levels') + 
  ylab('Salary')

# Visualize with More resolution
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() + 
  geom_point(aes(x = dataset$Level, y=dataset$Salary), colour = 'red') + 
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))), colour = 'blue') +
  ggtitle('Truth or Bluff (Regression model)') +
  xlab('Levels') + 
  ylab('Salary')


